src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -29,6 +29,17 @@ Copyright (c) .NET Foundation. All rights reserved.
                                         Publish
  
     The main publish entry point.
+
+    Extension hooks for injecting custom targets:
```

Comment:
@akoeplinger  Can you please provide more details about the architecture of the Mono extension? 
Please post a pointer if there's a doc about it, thanks.

I have the following main questions:
* Where do the Mono-specific targets live (ex: `CoreMonoAOTCompilation`)? 
* Why do we need the extensions in the SDK? How are they expected to be used by the Mono build? 
   Can it be handled via `Before/AfterTargest` in Mono-specific files?